expr/expressions_test.go

```diff
@@ -418,3 +419,37 @@ func TestRoundTripExtendedExpression(t *testing.T) {
 		assert.Truef(t, pb.Equal(&ex, out), "expected: %s\ngot: %s", &ex, out)
 	}
 }
+
+func TestCastVisit(t *testing.T) {
+	var builder = plan.NewBuilderDefault()
+	castExpr := expr.MustExpr(builder.GetExprBuilder().Cast(builder.GetExprBuilder().Wrap(
+		expr.NewLiteral[float64](12.0, true)),
+		&types.Float64Type{Nullability: types.NullabilityRequired}).FailBehavior(
+		types.BehaviorThrowException).BuildExpr())
+
+	type relationTestCase struct {
+		name            string
+		rewriteFunction func(rex expr.Expression) expr.Expression
+		want            string
+	}
+	testCases := []relationTestCase{
+		{"no change", func(ex expr.Expression) expr.Expression { return ex },
+			"literal:{fp64:12 nullable:true}"},
+		{"changed", func(ex expr.Expression) expr.Expression {
+			lit, err := expr.NewLiteral[float64](16.0, true)
+			if err != nil {
```

Comment:
nit: Four line of if can be replaced with using require
```suggestion
			require.NoError(t, err)
```

Reply:
done